recipe/meta.yaml

```diff
@@ -1,106 +1,279 @@
+        - _openmp_mutex                                       # [linux]
+        - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.12") }}  # [osx and x86_64]
+        - _x86_64-microarch-level >=2                         # [x86_64_opt == "v2"]
+        - _x86_64-microarch-level >=3                         # [x86_64_opt == "v3"]
+
+    test:
+      commands:
+        # clients now return 1 for after printing help...
+        # see https://github.com/ggerganov/llama.cpp/pull/7675
+        # see https://github.com/ggerganov/llama.cpp/blob/b3131/examples/main/main.cpp#L120-L127
+        - llama-cli --help   || true               # [not win]
+        - llama-cli --help   || cmd /c "exit /B 0" # [win]
+        - llama-server --help || true               # [not win]
+        - llama-server --help || cmd /c "exit /B 0" # [win]
+        - test -f $PREFIX/include/llama.h    # [unix]
+        - test -f $PREFIX/bin/llama-cli       # [unix]
+        - test -f $PREFIX/bin/llama-server   # [unix]
+        - test -f $PREFIX/lib/libllama.so    # [linux]
+        - test -f $PREFIX/lib/libllama.dylib # [osx]
+        - if not exist %PREFIX%/Library/include/llama.h exit 1  # [win]
+        - if not exist %PREFIX%/Library/lib/llama.lib exit 1    # [win]
+        - if not exist %PREFIX%/Library/bin/llama.dll exit 1    # [win]
+        - if not exist %PREFIX%/Library/bin/llama-cli.exe exit 1      # [win]
+        - if not exist %PREFIX%/Library/bin/llama-server.exe exit 1   # [win]
+
+    about:
+      home: https://github.com/ggerganov/llama.cpp
+      summary: LLM inference in C/C++
+      description: |
+        Inference of Meta's LLaMA model (and others) in pure C/C++
+      license: MIT
+      license_family: MIT
+      license_file: LICENSE
+      dev_url: https://github.com/ggerganov/llama.cpp
+      doc_url: https://github.com/ggerganov/llama.cpp
+
+    extra:
+      recipe-maintainers:
+        - sodre
+        - cbouss
+        - jnoller
+
+  - name: llama.cpp-tools
+    script: build-llama-cpp-tools.sh   # [not win]
+    script: bld-llama-cpp-tools.bat  # [win]
+    version: {{ version }}
+
+    build:
+      entry_points:
+        - llama-convert-hf-to-gguf = llama_cpp_tools.convert_hf_to_gguf:main
+        - llama-convert-llama-ggml-to-gguf = llama_cpp_tools.convert_llama_ggml_to_gguf:main
+        - llama-convert-lora-to-gguf = llama_cpp_tools.convert_lora_to_gguf:main
+        - llama-lava-surgery = llama_cpp_tools.examples.llava.llava_surgery:main
+        - llama-lava-surgery_v2 = llama_cpp_tools.examples.llava.llava_surgery_v2:main
+        - llama-convert-image-encoder-to-gguf = llama_cpp_tools.examples.llava.convert_image_encoder_to_gguf:main
+      skip: True # [py<39]
+      number: {{ build_number }}
+
+    requirements:
+      host:
+        - python
+        - poetry-core >=1.0.0
+        - pip
+      run:
+        - python
+        # requirements/requirements-convert_legacy_llama.txt
```

Comment:
ok, thanks. I still can't see how this list matches up with [here](https://github.com/ggerganov/llama.cpp/blob/b3989/requirements/requirements-convert_legacy_llama.txt), though. If there's an adaptation, could be helpful to document it for future updates.

Reply:
@danpetry 🤦 I completely missed that I had some of gguf-py's deps in the list still. Checking/fixing

Reply:
should be done in https://github.com/AnacondaRecipes/llama.cpp-feedstock/pull/11/commits/17745b6af25fa6a16a8056ca66c29dcd8cdaeedd

Reply:
Sorry I'm still not following. according to requirements/requirements-convert_legacy_llama.txt I think it should be
```
numpy~=1.26.4
sentencepiece~=0.2.0
transformers>=4.45.1,<5.0.0
gguf>=0.1.0
protobuf>=4.21.0,<5.0.0
```
?